docs/path-resolution.rst

```diff
@@ -0,0 +1,690 @@
+       CLI paths are platform-specific.
+       The same path may be interpreted differently on different systems.
+
+       .. code-block:: shell
+           :caption: Windows
+
+           solc.exe C:\project\token.sol &REM source unit name: C:/project/token.sol
+           solc.exe /project/token.sol   &REM source unit name: /project/token.sol
+
+       .. code-block:: bash
+           :caption: Linux
+
+           solc C:\project\token.sol # source unit name: C:projecttoken.sol
+           solc /project/token.sol   # source unit name: /project/token.sol
+
+#. **Standard JSON (as content)**
+
+   An alternative way to compile your project is to use the ``--standard-json`` option and provide
+   a JSON file containing all of your source code:
+
+   .. code-block:: json
+
+       {
+           "language": "Solidity",
+           "sources": {
+               "contract.sol": {
+                   "content": "import \"./util.sol\";\ncontract C {}"
+               },
+               "util.sol": {
+                   "content": "library Util {}"
+               },
+               "/usr/local/dapp-bin/token.sol": {
+                   "content": "contract Token {}"
+               }
+           },
+           "settings": {"outputSelection": {"*": { "*": ["metadata", "evm.bytecode"]}}}
+       }
+
+   The ``sources`` dictionary specifies the initial content of the virtual filesystem and you
+   can use source unit names directly there.
+   They do not undergo any extra translation or normalization.
+
+   The path to the JSON file does not affect the path resolution in any way.
+   In fact, it is common to supply it on the standard input in which case it does not have a path at all.
+
+   .. note::
+
+       When using ``--standard-json`` you cannot provide additional source files as command-line
+       arguments but it does not mean that the compiler will not load any extra files from disk.
+       If a contract imports a file that is not present in ``sources``, the compiler will use the
+       file loader as in any other situation, which may result in the source being read from disk
+       (or provided by the callback when using the JavaScript interface).
+
+#. **Standard JSON (as URL)**
+
+   When using :ref:`Standard JSON interface <compiler-api>` it is possible to tell the compiler to
+   use the file loader to obtain the content:
+
+   .. code-block:: json
+
+       {
+           "language": "Solidity",
+           "sources": {
+               "/usr/local/dapp-bin/token.sol": {
+                   "urls": ["/projects/mytoken.sol"]
+               }
+           },
+           "settings": {"outputSelection": {"*": { "*": ["metadata", "evm.bytecode"]}}}
+       }
+
+   The value specified in ``urls`` does not affect the source unit name and is not included in
+   contract metadata.
+   It is only passed to the file loader and used to locate the file.
+
+   As the name of the attribute implies, the value could be an URL if supported by the loader.
```

Comment:
```suggestion
   As the name of the attribute implies, the value could be a URL if supported by the loader.
```

Reply:
Maybe clarify: "http URLs"?

Reply:
Fixed.

> Maybe clarify: "http URLs"?

It's not just HTTP thought. Any protocol that the callback can handle is allowed. For example Remix can now handle IPFS.